version: '3.8'

services:
  nginx:
    image: nginx:1.27.0
    ports:
      - "8000:8000"
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./nginx/uwsgi_params:/etc/nginx/uwsgi_params:ro
      - static_volume:/static
    depends_on:
      - django

  mysql:
    image: mysql:8.0
    command: --default-authentication-plugin=mysql_native_password
    ports:
      - "3306:3306"
    env_file:
      - ./mysql/.env
    volumes:
      - mysql_data:/var/lib/mysql
      - ./sql:/docker-entrypoint-initdb.d
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: "20s"
      retries: 10

  django:
    build:
      context: ./django
      dockerfile: Dockerfile
    command: uwsgi --socket :8001 --module testerRecruting.wsgi --py-autoreload 1 --logto /tmp/mylog.log
    volumes:
      - ./src:/code
      - static_volume:/static
    expose:
      - "8001"
    env_file:
      - ./django/.env
    depends_on:
      mysql:
        condition: service_healthy
    logging:
      driver: "json-file"
      options:
        max-size: "10m"
        max-file: "3"

volumes:
  mysql_data:
  static_volume:
